android/beagle/src/test/java/br/com/zup/beagle/android/context/ContextActionExecutorTest.kt

```diff
@@ -164,4 +168,37 @@ class ContextActionExecutorTest : BaseAsyncActionTest() {
         verify(exactly = 1) { asyncAction.onActionStarted() }
         assert(onActionStartedWasCalled())
     }
+
+    @DisplayName("When execute Actions")
+    @Nested
+    inner class ExecuteActions() {
+
+        @DisplayName("Then should report the actions if it's action analytics")
+        @Test
+        fun testExecuteActionsReportAction() {
+            // Given
+            val eventId = "onChange"
+            val analyticsViewModel = mockk<AnalyticsViewModel>()
+            val value = PersonTest(name = NAME)
+            val analyticsValue = "onChange"
+            val actionAnalytics = mockk<ActionAnalytics>()
+            every { actionAnalytics.execute(any(), view) } just Runs
+            every { rootView.generateViewModelInstance<AnalyticsViewModel>() } returns analyticsViewModel
+            every { analyticsViewModel.createActionReport(rootView, any(), any(), any()) } just Runs
+            // When
+            contextActionExecutor.executeActions(
+                rootView,
+                view,
+                sender,
+                listOf(action, actionAnalytics),
+                ContextData(eventId, value),
+                analyticsValue
+            )
+
+            // Then
+            verify(exactly = 1) { action.execute(rootView, view) }
+            verify(exactly = 1) { actionAnalytics.execute(rootView, view) }
+            verify(exactly = 1) { analyticsViewModel.createActionReport(rootView, view, actionAnalytics, analyticsValue) }
```

Comment:
What do you think besides testing if the methods were called, do you also test their order?
```suggestion

            verifyOrder { 
                action.execute(rootView, view)
                actionAnalytics.execute(rootView, view)
                analyticsViewModel.createActionReport(rootView, view, actionAnalytics, analyticsValue)
            }
```

Reply:
Nice suggestion!